drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -227,9 +267,11 @@ static int adc_smartbond_read(const struct device *dev,
 	int error;
 	struct adc_smartbond_data *data = dev->data;
 
+	gpadc_smartbond_pm_policy_state_lock_get(data);
 	adc_context_lock(&data->ctx, false, NULL);
 	error = start_read(dev, sequence);
 	adc_context_release(&data->ctx, error);
+	gpadc_smartbond_pm_policy_state_lock_put(data);
```

Comment:
Though it will not trigger issues, this is redundant as policy constraints will be released from within ISR both for asynchronous and non-asynchronous APIs.

Reply:
Removed